pynxtools_stm/nomad/test_nomad_examples.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Test for NOMAD examples in MPES reader plugin."""
+
+import pytest
+
+try:
+    from nomad.parsing.parser import ArchiveParser
+    from nomad.datamodel import EntryArchive, Context
+except ImportError:
+    pytest.skip(
+        "Skipping NOMAD example tests because nomad is not installed",
+        allow_module_level=True,
+    )
+
+from pynxtools.testing.nomad_example import (
+    get_file_parameter,
+    parse_nomad_examples,
+    example_upload_entry_point_valid,
+)
+
+from pynxtools_mpes.nomad.entrypoints import mpes_example
+
+
+@pytest.mark.parametrize(
+    "mainfile", get_file_parameter("src/pynxtools_mpes/nomad/examples")
+)
+def test_parse_nomad_examples(mainfile):
+    """Test if NOMAD examples work."""
+    parse_nomad_examples(mainfile)
```

Comment:
`parse_nomad_examples` returns `archive.m_to_dict` now, so if you want, you can use
```suggestion
    archive_dict = parse_nomad_examples(mainfile)
```
and run further tests on that dict that are specific to the stm example(s).

Reply:
You can review newly added test_nomad_examples.py.